rmpv/src/ext/de.rs

```diff
@@ -534,6 +596,98 @@ impl<'de> Deserializer<'de> for &'de ValueRef<'de> {
     }
 }
 
+enum ExtData<'de> {
+    Owned(Vec<u8>),
+    Borrowed(&'de [u8])
+}
+
+struct ExtDeserializer<'de> {
+    tag: Option<i8>,
+    data: Option<ExtData<'de>>,
+}
+
+impl<'de> ExtDeserializer<'de> {
+    fn new_owned(tag: i8, data: Vec<u8>) -> Self {
+        ExtDeserializer {
+            tag: Some(tag),
+            data: Some(ExtData::Owned(data)),
+        }
+    }
+
+    fn new_ref(tag: i8, data: &'de [u8]) -> Self {
+        ExtDeserializer {
+            tag: Some(tag),
+            data: Some(ExtData::Borrowed(data)),
+        }
+    }
+}
+
+impl<'de> SeqAccess<'de> for ExtDeserializer<'de> {
+    type Error = Error;
+
+    fn next_element_seed<T>(&mut self, seed: T) -> Result<Option<T::Value>, Error>
+    where
+        T: DeserializeSeed<'de>,
+    {
+        if self.tag.is_some() || self.data.is_some() {
+            return Ok(Some(seed.deserialize(self)?));
+        }
+
+        Ok(None)
+    }
+}
+
+impl<'a, 'de: 'a> Deserializer<'de> for ExtDeserializer<'de> {
+    type Error = Error;
+
+    #[inline]
+    fn deserialize_any<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+        where V: Visitor<'de>
+    {
+        Err(de::Error::invalid_type(de::Unexpected::Other("non tuple struct"), &"Ext tuple struct"))
+    }
+
+    fn deserialize_tuple_struct<V>(self, _name: &'static str, _len: usize, visitor: V) -> Result<V::Value, Self::Error>
+        where V: Visitor<'de>
+    {
+        visitor.visit_seq(self)
+    }
+
+    forward_to_deserialize_any! {
+        bool u8 u16 u32 u64 i8 i16 i32 i64 f32 f64 char str string unit option
+        seq bytes byte_buf map unit_struct newtype_struct
+        struct identifier tuple enum ignored_any
+    }
+}
+
+impl<'a, 'de: 'a> Deserializer<'de> for &'a mut ExtDeserializer<'de> {
```

Comment:
For future code readers, could you add a small comment mentioning this implementation is used only inside the `SeqAccess` impl (as opposed to the above, which is used for general deserialization)?